### DISCUSSION TITLE
Cumulative Histogram in log2metrics

### DISCUSSION THREAD

**Discussion by szibis** [vectordotdev/vector#11920] [General]
Is there a way or new feature is needed to add histogram with buckets ranges in tags annotation just like implemented here - also with cumulative option ?? https://github.com/influxdata/telegraf/blob/master/plugins/aggregators/histogram/README.md

**Comment by jszwedko** [comment]
Hi @szibis !

You should be able to generate a histogram metric with `log_to_metric`. See the example here: https://vector.dev/docs/reference/configuration/transforms/log_to_metric/#examples-histogram-distribution

That just publishes a new sample for each incoming log event. The bucketing then happens in the sink. See the Prometheus exporter sink configuration for example: https://vector.dev/docs/reference/configuration/sinks/prometheus_exporter/#buckets

I could see us enhancing the `aggregate` transform to make an aggregated histogram earlier than the sink, including the buckets and cumulative option, if you wanted to open a feature request.

Let me know if this helps!

**Reply by szibis** [reply]
Yeah, that would be great.

**Reply by jszwedko** [reply]
Sounds good, would you want to open a feature request if adding it to the `aggregate` transform would be useful to you?

**Reply by szibis** [reply]
If I can achieve similar to this in Telegraf aggregation it wi be very useful.

**Reply by jszwedko** [reply]
Yeah, that makes sense. I'll leave it to you to open the feature request if you want it. You can do so here: https://github.com/vectordotdev/vector/issues/new?assignees=&labels=type%3A+feature&template=feature.yml
